lib/platform/github/index.js

```diff
@@ -913,16 +915,42 @@ async function getPr(prNo) {
       pr.canMerge = true;
     }
     if (pr.mergeable_state === 'dirty') {
-      logger.debug('PR state is dirty so unmergeable');
+      logger.debug({ prNo }, 'PR state is dirty so unmergeable');
       pr.isUnmergeable = true;
     }
     if (pr.commits === 1) {
-      // Only one commit was made - must have been renovate
-      logger.debug('Only 1 commit in PR so rebase is possible');
-      pr.canRebase = true;
+      if (config.gitAuthor) {
+        // Check against gitAuthor
+        logger.debug('Checking all commits');
+        try {
+          const commitAuthorEmail = (await get(
+            `repos/${config.parentRepo ||
+              config.repository}/pulls/${prNo}/commits`
+          )).body[0].commit.author.email;
+          const { address: gitAuthorEmail } = addrs.parseOneAddress(
+            config.gitAuthor
+          );
+          if (commitAuthorEmail === gitAuthorEmail) {
+            logger.debug({ prNo }, '1 commit matches gitAuthor = can rebase');
+            pr.canRebase = true;
+          } else {
+            logger.debug(
+              { prNo },
+              'PR commit is not by gitAuthor = cannot rebase'
```

Comment:
This particulate debug message doesn't really make sense to me. The commit cannot be rebased because the first commit on the PR is not authored by Renovate, right?

Reply:
Yes, it's the case I'm trying to support - where someone has force pushed over Renovate's commit. In this case we want to abandon any rebasing/recreating as otherwise we'll essentially force push over their force push.

Reply:
Can the debug message be written to make that more apparent? The current wording doesn't really provide enough detail to understand what "not by gitAuthor" really means.